### DISCUSSION TITLE
DAI 2024-2025 - Practical work 1 - Florian Chollet and Alexandre Delétraz

### DISCUSSION THREAD

**Discussion by luma2010** [heig-vd-dai-course/heig-vd-dai-course#369] [Show and tell]
Notre idée est d'utilisé picocli afin de crée un éditeur de fichier contenant des variables. Exemple nous avons une varible "counter=100", notre idée est de faire un programme qui permettrait de changer counter=100 à par exemple counter=50 (ou de rajouter des variables) avec une syntaxe similaire à "java -jar NomDuProjet NomDuFichier.txt --NomOption ..."

**Comment by ludelafo** [comment]
Juste pour être au clair avec votre idée, pourriez-vous nous préciser quelles sont les commandes que vous souhaitez implémenter pour votre CLI avec des exemples ?

En dehors de ça, cela nous semble OK, merci !

**Reply by luma2010** [reply]
pour le moment nous en avions 3 :
-a, --add pour rajouter une variable
-d, --delete pour supprimer une variable
-m --modify pour modifier une variables

**Reply by ludelafo** [reply]
Super ! C'est OK pour moi. @hadrylouis, tu valides aussi ?

Je préfère être bien précis pour éviter une quelconque confusion si vous devez fouiller dans la documentation de picocli :

picocli parle d'[options](https://picocli.info/#_options) lorsqu'on utilise des éléments tels que ceux que vous venez de nous décrire (`--add`, `--delete`, `--modify` dans leur version longue ou `-a`, `-d`, `-m` dans leur version courte).

Ce que vous souhaitez utiliser, ce sont des [sous-commandes](https://picocli.info/#_subcommands). Ces sous-commandes permettront d'effectuer des actions sur vos fichiers. Nous aurons donc les sous-commandes `add`, `delete`, `modify` qui, elles, pourraient prendre des options et/ou des [paramètres](https://picocli.info/#_positional_parameters) obligatoires ou optionnels pour passer des arguments à vos sous-commandes.

Vous avez pu explorer tout cela au travers du contenu pratique du cours _"Java, IntelliJ IDEA and Maven"_ qui vous a demandé de mettre en place un CLI avec deux sous-commandes `hello` et `goodbye` et ses options/paramètres associés. Vous pouvez vous en inspirer si besoin.

J'espère avoir été clair dans mon discours, n'hésitez pas à revenir vers nous si besoin.

**Comment by luma2010** [comment]
Voila le rendu de notre application :
Lien du repo : https://github.com/luma2010/heig-vd-dai-projet-pratique1.git
Hash du dernier commit : 9b483294ea819c96dc01448ffa6cad2d99faf064
